### PR TITLE
Refactor/dispute resolver

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
- updates Dispute resolver Aliases to be optional

- updates dispute resolver template types
   - answer.description now mandatory
   - answer.id now saved as hex-string
   - category now mandatory
   - added version field
   - added arbitrableAddress, arbitrableChainId, arbitratorAddress, arbitratorChainId fields

---

## PR-Codex overview
This PR updates various components in the Resolver feature, focusing on improving form validation, context handling, and component styling.

### Detailed summary
- Updated form validation logic for voting options
- Refactored context handling for new dispute data
- Improved component styling by removing unnecessary CSS
- Added new fields and logic for dispute templates

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

- **New Features**
  - Introduced versioning to dispute templates for better tracking and management.
  - Enhanced dispute templates with required category descriptions and IDs.
  - Strengthened input validation for user voting options by requiring both title and description.

- **Refactor**
  - Improved context handling for new disputes to handle undefined initial values.
  - Simplified the styling logic by removing the `StyledLabel` component.
  - Updated internal logic to convert answer IDs to hexadecimal representation.

### CODE REVIEW

**Review by kemuru — DISMISSED** [review]

File: web/src/context/NewDisputeContext.tsx [hunk 1]
```diff
@@ -62,14 +65,13 @@ const initialDisputeData: IDisputeData = {
   title: "",
```
Comment (alcercu) [inline web/src/context/NewDisputeContext.tsx:65]: I have been preferring the following solution over creating an object for context initialization:
```typescript
const NewDisputeContext = createContext<INewDisputeContext | undefined>(undefined)

export const useNewDisputeContext = () => {
  const context = useContext(NewDisputeContext);
  if (!context) {
    throw new Error("Context Provider not found.");
  }
  return context;
};
```
wdyt?

Comment (Harman-singh-waraich) [reply web/src/context/NewDisputeContext.tsx:65]: hmm, i agree with the check. 
Also reduces the code size 👍

**Review by jaybuidl — CHANGES_REQUESTED** [review]
Let's remove:
- numberOfJurors
- arbitrationCost
- arbitrableAddress

**Review by jaybuidl — APPROVED** [review]

File: web/src/context/NewDisputeContext.tsx [hunk 2]
```diff
@@ -130,7 +128,17 @@ const constructDisputeTemplate = (disputeData: IDisputeData) => {
...
+    answer.id = "0x" + BigInt(answer.id).toString(16);
+  }
   if (!isUndefined(baseTemplate.policyURI) && baseTemplate.policyURI === "") delete baseTemplate.policyURI;
 
+  // in future if we support chain switching these would need to be calculated from connected 'chainId'
+  baseTemplate.arbitrableAddress = disputeResolverAddress[DEFAULT_CHAIN];
+  baseTemplate.arbitrableChainID = DEFAULT_CHAIN.toString();
+  baseTemplate.arbitratorAddress = klerosCoreAddress[DEFAULT_CHAIN];
+  baseTemplate.arbitratorChainID = DEFAULT_CHAIN.toString();
+
```
[the remark this answers is not in the document]
Comment (jaybuidl) [reply web/src/context/NewDisputeContext.tsx]: @coderabbitai is this still relevant?

**Pull request merged**